starter-code/main.js

```diff
@@ -16,54 +16,235 @@ mongoClient.connect(url, (error, db) => {
+            db.collection('companies').find({}, { "relationships.person.permalink": "david-ebersman" }, { name: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "14":
+            //List by name the competitors of Facebook
+            db.collection('companies').find({}, { name: "Facebook" }, { "competitions.competitor.name": 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "15":
+            //Names of the companies that has "social-networking" in tag-list 
+            db.collection('companies').find({}, { tag_list: { $regex: /social-networking/ } }, { name: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "16":
+            //How many companies that has "social-network" in tag-list and founded between 2002 and 2016 inclusive 
+            db.collection('companies').find({}, { $and: [{ tag_list: { $regex: /social-networking/ } }, { founded_year: { $gte: 2002 } }, { founded_year: { $lte: 2016 } }] }).count((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
+          case "17":
+            //Names and locations of companies that have offices in London
+            db.collection('companies').find({}, { "offices.city": { $in: ["London"] } }, { name: 1 }).toArray((error, result) => {
+              if (error) {
+                console.log(error);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              } else {
+                console.log(result);
+                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+              }
+            })
+            break;
```

Comment:
```suggestion
          case "17":
            //Names and locations of companies that have offices in London
            db.collection('companies').find({}, {name: 1, offices: 1, _id: 0}).toArray((error, result) => {
              if (error) {
result.forEach(value => {
                  try{
                    if(value.offices[0].city == 'London')console.log(value.name);
                  }catch(e){};
                });                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
              } else {
                console.log(result);
                rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
              }
            })
            break;
```
Se refería al campo city.